### PR TITLE
Attribute model and translations morphMany relation

### PR DESCRIPTION
**Pull request by massimomegistus**
TranslatableBehavior with a "translations" morphMany relation to the new Attribute model, so you can use the Eloquent eager loading, for example:

```php
public function scopeList($query)
     {
         return $query->with([
           'translations',
           'featured_image',
           'featured_image.translations',
         ]);
     }
```

### CODE REVIEW

**Comment by w20k** [comment]
@massimomegistus could you also add unit tests to a `TranslatableModelTest.php`?

**Comment by massimomegistus** [comment]
@w20k unfortunately I have an issue running phpunit from rainlab/translate folder . I get this error
Cannot open file "/var/www/public/octest/plugins/rainlab/translate/../../../tests/bootstrap.php"
Any idea? Thanks for the support.

**Comment by w20k** [comment]
@massimomegistus the easiest way would be to add `translate-plugin` to a live|active project with everything setup. `Composer update` and run tests from the root directory, like so: `php ./vendor/bin/phpunit plugins/rainlab/translate/tests/unit`.

**Comment by massimomegistus** [comment]
It works, thank you! I'll add the unit test very soon (by the end of the week)

**Comment by w20k** [comment]
Cool, thanks! Once you're done, will revise it! ;)

File: behaviors/TranslatableModel.php [hunk 1]
```diff
@@ -208,13 +227,11 @@ protected function loadTranslatableData($locale = null)
             return $this->translatableAttributes[$locale] = [];
         }
 
-        $obj = Db::table('rainlab_translate_attributes')
-            ->where('locale', $locale)
-            ->where('model_id', $this->model->getKey())
-            ->where('model_type', get_class($this->model))
-            ->first();
+        $obj = $this->model->translations->first(function ($value, $key) use($locale) {
```
Comment (bennothommo) [inline behaviors/TranslatableModel.php]: @massimomegistus Please add a space between `use` and `(`:
```suggestion
        $obj = $this->model->translations->first(function ($value, $key) use ($locale) {
```

File: behaviors/TranslatableModel.php [hunk 2]
```diff
@@ -208,13 +227,11 @@ protected function loadTranslatableData($locale = null)
...
+            return $value->attributes['locale'] === $locale;
+        });
 
-        $result = $obj ? json_decode($obj->attribute_data, true) : [];
+        $result = $obj ? json_decode($obj['attribute_data'], true) : [];
```
Comment (bennothommo) [inline behaviors/TranslatableModel.php]: @massimomegistus Would you mind letting us know the reason for changing this line? The original line should've worked.

File: behaviors/TranslatableModel.php (hunk 1 above)
Comment (mjauvin) [reply behaviors/TranslatableModel.php]: The first() method's arguments should be ($key, $value) ...

ref. https://octobercms.com/docs/services/collections#method-first

**Comment by mjauvin** [comment]
@massimomegistus I like your implementation!

File: behaviors/TranslatableModel.php (hunk 1 above)
Comment (mjauvin) [reply behaviors/TranslatableModel.php]: Actually, looks like the documentation referenced above is wrong the correct order for the callable arguments in the code is ($value, $key):

ref. `vendor/laravel/framework/src/Illuminate/Support/Arr.php` (line 175)

Comment (bennothommo) [reply behaviors/TranslatableModel.php]: @mjauvin I was gonna say... I thought it was always $value, $key. :stuck_out_tongue:

Comment (bennothommo) [reply behaviors/TranslatableModel.php]: Fixed in docs: https://github.com/octobercms/docs/commit/0ff464042c3630ff0026710c2f7376fd9685ea0e

File: behaviors/TranslatableModel.php (hunk 2 above)
Comment (massimomegistus) [reply behaviors/TranslatableModel.php]: @bennothommo there's no reason, In fact. That's a useless change.

Comment (bennothommo) [reply behaviors/TranslatableModel.php]: @massimomegistus Feel free to restore it for us :)

**Comment by massimomegistus** [comment]
@mjauvin thank you so much :)

**Comment by massimomegistus** [comment]
Hello @w20k, yesterday I included the testGetTranslationValueEagerLoading in TranslatableModelTest

**Comment by w20k** [comment]
@massimomegistus cool-cool, thanks! Already tested unit test ;) Works like a charm! 
Screenshot 2019-10-24 at 14 05 57

**Comment by massimomegistus** [comment]
@w20k Great! Good news are the best!

File: behaviors/TranslatableModel.php [hunk 3]
```diff
@@ -21,6 +21,25 @@
...
+    {
+        parent::__construct($model);
+
+        $model->morphMany['translations'] = [
+            'RainLab\Translate\Models\Attribute',
+            'name' => 'model'
+        ];
+    }
+
+    public function scopeTransWith($query)
```
Comment (LukeTowers) [inline behaviors/TranslatableModel.php]: What's the point of this scope? Why wouldn't developers just use `with(['translations'])` by itself?

Comment (massimomegistus) [reply behaviors/TranslatableModel.php]: Hi @LukeTowers, you are right. scopeTransWith could be chained to other scopes, but "with" option is much more flexible. I'll remove scopeTransWith without any regrets.

Comment (massimomegistus) [reply behaviors/TranslatableModel.php]: @LukeTowers scopeTransWith method removed

**Comment by mjauvin** [comment]
Looks good to me, I'll test this in real context and would be ready to merge unless someone else finds a problem.

**Comment by mjauvin** [comment]
@hirisov can you test this PR and let us know if this addresses issue #502 ?

**Comment by massimomegistus** [comment]
Hi guys, in here again. Do you think this PR could me merged in the next release, or you see any drawback? Thanks!

**Comment by mjauvin** [comment]
Yes, we're only waiting for @hirisov to test this and confirm it's working and fixes his issue.

**Comment by mjauvin** [comment]
@bunker-condenast can you test this PR and confirm this is working as expected?

**Comment by bunker-condenast** [comment]
Hi @mjauvin , i'm already using it on two websites and it works fine

**Comment by mjauvin** [comment]
@w20k, we're missing your approval on this, care to approve/disprove?

**Pull request merged**